shoper.py

```diff
@@ -0,0 +1,135 @@
+import os
+import pandas as pd
+from operator import itemgetter
+from re import search
+
+
+def deactivate_products(database, dc_price, replacement_price):
+    """
+    Deactivates products set with exact price - e.g. 107.99
+    Products are kept because some of them have been withdrawn,
+    yet they might come back
+    
+    csv_file: .csv file to work with
+    dc_price: exact price that triggers deactivation of product
+    column_types: column types ensuring proper encoding of file
+    """
+    print("\nDeactivated following products:")
+    
+    for row in range(database.shape[0]):
+        
+        if database['price'][row]:
```

Comment:
Oba te ify mają ten sam blok kodu. Conditionale można zatem zwinąć.

```suggestion
        if database['price'][row] or pd.isnull(database.loc[row, 'price]):
```